Update service-calls.markdown Use response data in templates

Mariusthvdb opened the pull request:
convert Yaml to JSON
directly related to the new service weather.get_forecasts
see: https://github.com/home-assistant/core/pull/102534

## Proposed change
<!-- 
    Describe the big picture of your changes here to communicate to the
    maintainers why we should accept this pull request. If it fixes a bug
    or resolves a feature request, be sure to link to that issue in the 
    additional information section.
-->



## Type of change
<!--
    What types of changes does your PR introduce to our documentation/website?
    Put an `x` in the boxes that apply. You can also fill these out after
    creating the PR.
-->

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [ ] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information
<!--
    Details are important, and help maintainers processing your PR.
    Please be sure to fill out additional details, if applicable.
-->

- Link to parent pull request in the codebase: https://github.com/home-assistant/core/pull/102534
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: fixes #

## Checklist
<!--
    Put an `x` in the boxes that apply. You can also fill these out after
    creating the PR. If you're unsure about any of them, don't hesitate to ask.
    We're here to help! This is simply a reminder of what we are going to look
    for before merging your code.
-->

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

Mariusthvdb committed:
Update service-calls.markdown Use response data in templates

convert Yaml to JSON

FletcherAU commented on source/_docs/scripts/service-calls.markdown:
```diff
@@ -176,6 +176,86 @@ data:
 
 {% endraw %}
 
+### Use response data in templates: convert Yaml to JSON
```
```suggestion
### Use response data in templates: convert YAML to JSON
```
YAML is an acronym

FletcherAU commented on source/_docs/scripts/service-calls.markdown:
```diff
@@ -176,6 +176,86 @@ data:
+
+ Please follow this approach to be able to test run your templates based on the response:
+
+ 1 enter service in `/developer-tools/service` eg:
```
```suggestion
1. Enter service in `/developer-tools/service` eg:
```

FletcherAU commented on source/_docs/scripts/service-calls.markdown:
````diff
@@ -176,6 +176,86 @@ data:
+ ```
+ {% endraw %}
+
+ 2 copy the exact and complete response
````
```suggestion
2. Copy the exact and complete response
```

FletcherAU commented on source/_docs/scripts/service-calls.markdown:
```diff
@@ -176,6 +176,86 @@ data:
+
+ 2 copy the exact and complete response
+
+ 3 paste that in an (online) Yaml-JSON converter and copy result
```
```suggestion
3. Paste that in an (online) Yaml-JSON converter and copy result.
```

FletcherAU commented on source/_docs/scripts/service-calls.markdown:
```diff
@@ -176,6 +176,86 @@ data:
+
+ 3 paste that in an (online) Yaml-JSON converter and copy result
+
+ 4 paste that result in `/developer-tools/template` inside a setter:
```
```suggestion
4. Paste that result in `/developer-tools/template` inside a setter:
```

FletcherAU commented on source/_docs/scripts/service-calls.markdown:
```diff
@@ -176,6 +176,86 @@ data:
+
+ {% endraw %}
+
+ 5 template to your liking, eg:
```
```suggestion
5. Template to your liking, eg:
```

FletcherAU reviewed (changes requested):
Slight tweaks to formatting

Mariusthvdb committed:
Update service-calls.markdown

Co-authored-by: Fletcher <me@fboyd.me>

Mariusthvdb committed:
Update service-calls.markdown

Co-authored-by: Fletcher <me@fboyd.me>

Mariusthvdb committed:
Update service-calls.markdown

Co-authored-by: Fletcher <me@fboyd.me>

Mariusthvdb committed:
Update service-calls.markdown

Co-authored-by: Fletcher <me@fboyd.me>

Mariusthvdb committed:
Update service-calls.markdown

Co-authored-by: Fletcher <me@fboyd.me>

Mariusthvdb committed:
Update service-calls.markdown

Co-authored-by: Fletcher <me@fboyd.me>

Mariusthvdb commented:
Thx

Mariusthvdb committed:
add Onlineyamltools as example service

for converting. Have a reliable service directly linked, saves the user some extra searching

frenck commented on source/_docs/scripts/service-calls.markdown:
```diff
@@ -176,6 +176,86 @@ data:
 
 {% endraw %}
 
+### Use response data in templates: convert YAML to JSON
```
This title doesn't make sense to me considering the context below.

What is the actual thing this section achieves and/or tries to document?

Mariusthvdb replied:
goal is to explain in the HA documentation how to use the response data in ones templates, and how this can be done converting the presented Yaml to JSON and next, use that JSON data structure inside dv tools template

frenck commented on source/_docs/scripts/service-calls.markdown:
```diff
@@ -176,6 +176,86 @@ data:
 
+### Use response data in templates: convert YAML to JSON
+
+ If the response data is a larger data object, (like with the `weather.get_forecasts` service) you can not use the response directly in the template editor at `/developer_tools/template`.
```
- Leading space?
- This is not an object?
- Let's use My links instead of relative URL paths (it is unclear where to use those).

Mariusthvdb replied:
not sure how to use My links here, looking for an example

frenck commented on source/_docs/scripts/service-calls.markdown:
```diff
@@ -176,6 +176,86 @@ data:
+### Use response data in templates: convert YAML to JSON
+
+ If the response data is a larger data object, (like with the `weather.get_forecasts` service) you can not use the response directly in the template editor at `/developer_tools/template`.
+ The response needs to be converted to JSON to be able to do so.
```
That sounds wrong? Why?

JSON != Python

frenck reviewed (changes requested):
I've stopped reviewing, as I'm really unsure what this is trying to achieve.

The chapter looks like a mixed bag of things, without actually explaining what the goal is, what is being resolved, and why this all is needed?

../Frenck

Mariusthvdb committed:
Update service-calls.markdown

adapted text and spaces

Mariusthvdb committed:
add My links

Mariusthvdb commented:
believe to have answered your points and made the changes, marking ready for further review.
thanks

frenck commented on source/_docs/scripts/service-calls.markdown:
```diff
@@ -176,6 +176,87 @@ data:
 
+### Use response data in templates: convert YAML to JSON
+
+The response data is returned in Yaml, and can not be used as is in the template editor.
```
```suggestion
The response data is returned in YAML and can not be used as is in the template editor.
```

I'm not sure why this matters?

Mariusthvdb replied:
I added that because in the response after the new service was implemented, users did not understand what happened, or how they could use that response.
understanding the type of data is the beginning of the way to the answer, so seems important

frenck commented on source/_docs/scripts/service-calls.markdown:
```diff
@@ -176,6 +176,87 @@ data:
+### Use response data in templates: convert YAML to JSON
+
+The response data is returned in Yaml, and can not be used as is in the template editor.
+The response needs to be converted to JSON to be able to do so.
```
Why?

If we want to teach users, let's teach them how to read and use what we show, and not add additional formats into the mix.

Mariusthvdb replied:
believe this is the same question?
I merely tried to describe the full process of getting the response, and handling that response to finally get some format we can use in the templates.

as I see it, we need to describe the difference between the data in yaml format, and how to make  a json object of it to be used in templates

its not just some additional format, it is what Home Assistant returns, there is no way around that

Mariusthvdb committed:
Update source/_docs/scripts/service-calls.markdown

Co-authored-by: Franck Nijhof <frenck@frenck.nl>

klaasnicolaas commented:
If we do anything else with this PR, we would like to rebase it to the `current` branch.

./Klaas

Mariusthvdb commented:
Ofc, please do

klaasnicolaas commented:
You must rebase it 😉
